lib/dao.js

```diff
@@ -415,7 +415,9 @@ DataAccessObject.updateOrCreate = DataAccessObject.upsert = function upsert(data
         var update = data;
         var inst = data;
         if(!(data instanceof Model)) {
+          console.log('not instance', data);
           inst = new Model(data);
```

Comment:
Ok, I'll look into adding another option here: https://github.com/strongloop/loopback-datasource-juggler/blob/master/lib/model.js#L42
With that value turned on we skip the default value and defaultFn generation blocks.